src/transport.js

```diff
@@ -128,6 +129,15 @@ class TransportManager {
 
     parallel(createListeners, (err) => {
       if (err) {
+        // Error connecting to WebSocket
+        if (err.description && err.description.code === 'ENOTFOUND') {
+          const hostname = err.description.hostname
+
+          err = Object.assign(new Error(`WebSocket connection failed on ${hostname}`), {
```

Comment:
I don't think this pertains in the switch - which is transport agnostic. If anything, this should be emitted by the transport itself.

Reply:
Makes sense. I should probably close this PR and put this into [js-libp2p-websocket-star](https://github.com/libp2p/js-libp2p-websocket-star) then?

Reply:
Since the error is specific to websockets it makes sense for that to exist in js-libp2p-websocket-star and not here. I think there are two things to take into account here. 

1. This PR looks like you're just trying to get more meaningful errors logged/thrown when things break to get better information to developers. If that's the case, creating that error in websocket-star and letting it propagate through libp2p-switch to the user should be sufficient for the short term.

2. The larger issue this brings up is better error messaging and handling in general. Right now, a lot of errors coming out of switch via the transports aren't terribly helpful and can be really hard to debug. I think what I'd like to  see happen for this is to create the generic, extendable error types in https://github.com/libp2p/interface-connection and then have each transport catch and add meaningful metadata to those, before passing them on. This would enable switch, and other libp2p layers to handle the generic errors and take action accordingly, while giving users more detailed metadata from specific transports. If the error is meant to be fatal we can handle that appropriately. When the more common scenario of a bad connection occurs, we can simply terminate that connection and move on. Combining this with a move of switch to a state machine would be really helpful in making this a more resilient, debuggable system. cc: @diasdavid

Reply:
Thanks for the input @dryajov and @jacobheun. I'll close this one.

@jacobheun I'm going to open a more generic issue about error handling and daemon stability and it would be cool to have your second point there for an open discussion.